src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -562,7 +562,7 @@ internal bool HasPointerType
             {
                 if (_lazyType != null)
                 {
-                    return _lazyType.Value.DefaultType.IsPointerType();
+                    return _lazyType.Value.DefaultType.IsPointerOrFunctionPointer();
```

Comment:
>return _lazyType.Value.DefaultType.IsPointerOrFunctionPointer(); [](start = 20, length = 64)

Consider asserting that the value we get is consistent with what we would have inferred from syntax below. I think we already doing something like this for fields.